x-pack/plugins/reporting/server/plugin.ts

```diff
@@ -34,14 +39,36 @@ export class ReportingPlugin
 
   public setup(core: CoreSetup, plugins: ReportingSetupDeps) {
     // prevent throwing errors in route handlers about async deps not being initialized
-    core.http.registerRouteHandlerContext('reporting', () => {
+    core.http.registerRouteHandlerContext(PLUGIN_ID, () => {
       if (this.reportingCore.pluginIsStarted()) {
         return {}; // ReportingStart contract
       } else {
         return null;
       }
     });
 
+    core.uiSettings.register({
+      [UI_SETTINGS_CUSTOM_PDF_LOGO]: {
+        name: i18n.translate('xpack.reporting.pdfFooterImageLabel', {
+          defaultMessage: 'PDF footer image',
+        }),
+        value: null,
+        description: i18n.translate('xpack.reporting.pdfFooterImageDescription', {
+          defaultMessage: `Custom image to use in the PDF's footer`,
+        }),
+        type: 'image',
+        schema: schema.nullable(schema.byteSize({ max: '200kb' })),
+        category: [PLUGIN_ID],
```

Comment:
`schema` is required by types for `register`, however `validation` is what the client is looking at for validating that the size is appropriate